tactics/btermdn.ml

```diff
@@ -100,7 +101,13 @@ let constr_val_discr env sigma ts t =
     | Lambda _ -> Everything
     | Sort _ -> Label(SortLabel, [])
     | Evar _ -> Everything
-    | Case _ -> Everything (* Overapproximate wildly. TODO: be less brutal. *)
+    | Case (_, _, _, _, _, c, _) ->
+      begin
+        match decomp stack c with
+        | Label (GRLabel (ConstructRef _), _)  -> Everything (* over-approximating w.r.t. [fMATCH] *)
+        | Label _  -> Label(CaseLabel, c :: stack)
+        | _ as result -> result
```

Comment:
You should probably always return Everything here.

Reply:
I was just re-reading the code to understand the distinction. I suppose `Nothing` and `Everything` are only used inside this function since both are translated to `None` in the dnet. This got me very confused..

I now think the correct way to do this would be:

```OCaml
match decomp stack c with
| Label (GRLabel (ConstructRef _), _)  -> Everything (* over-approximating w.r.t. [fMATCH] *)
| Label _  | Nothing -> Label(CaseLabel, c :: stack)
| Everything -> Everything
```
What do you think?

Reply:
Yes, this is also correct.